src/method.rs

```diff
@@ -0,0 +1,198 @@
+                    }
+                }
+                None => Err(WMIError::ResultEmpty),
+            }
+        }
+    }
+
+    pub fn exec_class_method<MethodClass, In, Out>(
+        &self,
+        method: impl AsRef<str>,
+        in_params: In,
+    ) -> WMIResult<Option<Out>>
+    where
+        MethodClass: de::DeserializeOwned,
+        In: Serialize,
+        Out: de::DeserializeOwned,
+    {
+        let (method_class, _) = struct_name_and_fields::<MethodClass>()?;
+        self.exec_instance_method::<MethodClass, In, Out>(method, method_class, in_params)
+    }
+
+    pub fn exec_instance_method<MethodClass, In, Out>(
+        &self,
+        method: impl AsRef<str>,
+        object_path: impl AsRef<str>,
+        in_params: In,
+    ) -> WMIResult<Option<Out>>
+    where
+        MethodClass: de::DeserializeOwned,
+        In: Serialize,
+        Out: de::DeserializeOwned,
+    {
+        let (method_class, _) = struct_name_and_fields::<MethodClass>()?;
+        let serializer = VariantStructSerializer::new();
+        match in_params.serialize(serializer) {
+            Ok(field_map) => {
+                let field_map: HashMap<String, VARIANT> = field_map
+                    .into_iter()
+                    .filter_map(|(k, v)| match TryInto::<VARIANT>::try_into(v).ok() {
+                        Some(variant) => Some((k, variant)),
+                        None => None,
+                    })
+                    .collect();
+                let output =
+                    self.exec_method_native_wrapper(method_class, object_path, method, field_map);
+
+                match output {
+                    Ok(wbem_class_obj) => match wbem_class_obj {
+                        Some(wbem_class_obj) => Ok(Some(wbem_class_obj.into_desr()?)),
+                        None => Ok(None),
+                    },
+                    Err(e) => Err(e),
+                }
+            }
+            Err(e) => Err(WMIError::ConvertVariantError(e.to_string())),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::tests::fixtures::wmi_con;
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Deserialize)]
+    struct Win32_Process {
+        __Path: String,
+    }
+
+    #[derive(Serialize)]
+    struct CreateParams {
+        CommandLine: String,
+    }
+
+    #[derive(Serialize)]
+    struct TerminateParams {}
+
+    #[derive(Deserialize)]
+    struct TerminateOutput {}
+
+    #[derive(Deserialize)]
+    #[allow(non_snake_case)]
+    struct CreateOutput {
+        ProcessId: u32,
+    }
+
+    #[test]
+    fn it_exec_class_method() {
+        let wmi_con = wmi_con();
+        let in_params = CreateParams {
+            CommandLine: "notepad.exe".to_string(),
+        };
+
+        let out = wmi_con
+            .exec_class_method::<Win32_Process, CreateParams, CreateOutput>("Create", in_params)
+            .unwrap()
+            .unwrap();
+
+        assert!(out.ProcessId != 0);
```

Comment:
Kill the process in the end so that notepads don't keep spawning when running the tests 😅

Reply:
Done! Killing the process is an instance method, though, so since both tests were otherwise doing exactly the same thing after this change, I got rid of this one.